### ISSUE TITLE
Checkpoint Migration Issue / Clarification Request / Possible Improvement

### ISSUE THREAD

**Issue by jackcasey-visier** [signalfx/splunk-otel-collector#3960]
Hello! 😄 

# Summary

We're an organization that has been using Splunk Connect as our primary way to get our Kubernetes (on EKS) container logs into Splunk. With the upcoming depreciation, we're moving to the Splunk OpenTelemetry Collector.

So far it's worked seamlessly, other than one issue we've run into during the migration. After migrating, we're seeing duplicate events in Splunk. This has lead me to believe the Fluentd position files aren't being migrated or read as expected. I'm not clear on if this is an expected behaviour, or possibly an area we can improve with the migration :) 

For clarity, the migration I'm referencing is the functionality added here: https://github.com/signalfx/splunk-otel-collector/pull/843

# Recreation

To recreate this, we're running a basic pod in a sandbox we're calling `loggy` to print 1 log per second.

As expected, we can see the position file is marking last read position for the container logs: 

```
> cat /var/log/splunk-fluentd-containers.log.pos

<omitted>
/var/log/containers/loggy_default_loggy-aa608f729df8ee9d8247a5d9fa9953170ce29524a72ae5936e2521e0a2bf007a.log	000000000001fc9e	00000000000fba5c
```

At this point, we stop Splunk Client Connect and install the Splunk OpenTelemetry Collector. As a sanity check, I can see the init container completes without logging any errors:

```
> k logs splunk-otel-collector-agent-srmrk -c migrate-checkpoint

2023/11/23 00:25:16 Checkpoint migration completed
```

However, when checking in Splunk we see the previously collected events are now duplicated.

When looking into the otel logs, we can see this: 

```
> k logs splunk-otel-collector-agent-srmrk -c otel-collector

2023-11-23T00:25:18.176Z	info	service@v0.87.0/service.go:169	Everything is ready. Begin running and processing data.
2023-11-23T00:25:18.383Z	info	fileconsumer/file.go:160	Started watching file	{"kind": "receiver", "name": "filelog", "data_type": "logs", "component": "fileconsumer", "path": "/var/log/pods/default_loggy_44e6d6d2-0061-4ed0-91b8-4525ce52b84d/loggy/0.log"}
```

Something that stands out to me, is the fluentd position file uses the path `/var/log/containers/loggy...` where the otel logs are reading from `/var/log/pods/default_loggy...`

Interestingly, these are referencing both sides of a symlink: 

```
/var/log/containers/loggy_default_loggy-aa608f729df8ee9d8247a5d9fa9953170ce29524a72ae5936e2521e0a2bf007a.log -> /var/log/pods/default_loggy_44e6d6d2-0061-4ed0-91b8-4525ce52b84d/loggy/0.log
```

This leads me to believe the checkpoint migration is working, however the path being read is the non-linked one, causing the position to be missed. I haven't been able to confirm this as I haven't been able to find how to read the checkpoint files (any insight greatly appreciated)

# Next Steps? 

We're running most things with the defaults, so I suspect we're not the only ones that have run into this :) I've looked into configuration available in both fluentd and otel, but haven't found anything that could resolve this behaviour.

In summary, I'm hoping to get clarity on the following:

- Is this an expected behaviour? Are we missing something obvious to get around this?
- Would adding another migration step to migrate position of both the symlink+non-linked file make sense in this case? If so, I'm happy to contribute this change

Thank you! Other than this, the use of Splunk OpenTelemetry Collector has been butter smooth!

**Comment by jackcasey-visier** [comment]
Tagging @rockb1017 as you're likely the greatest mind on earth to weigh in on this 😅

**Comment by SBongers** [comment]
Any update on this? Our organization encountered the same issue having duplicate entries after using the checkpoint translation.

**Comment by atoulme** [comment]
This issue was fixed by https://github.com/signalfx/splunk-otel-collector/pull/3879 and released with the collector chart v0.91.0. We have tests to check migration is working with no duplicates here now: https://github.com/signalfx/splunk-otel-collector-chart/pull/1077

I am going to close this issue and ask that you please continue this discussion over a support case if it persists, as this is a better way to escalate quickly. Thank you for your report.

**Comment by SBongers** [comment]
@atoulme I'm still encountering some issues with the journald logs translation. This currently prevents the containers from being fully deployed. Where could I create such a support case. via splunk?
